src/test/objectstore/store_test.cc

```diff
@@ -3414,6 +3414,212 @@ TEST_P(StoreTest, OmapSimple) {
+        }
+        km[key].append(val);
+        b++;
+        k++;
+      }
+      ObjectStore::Transaction t;
+
+      // time omap_setkeys and queue_transaction
+      const utime_t start = ceph_clock_now();
+      t.omap_setkeys(cid, objects[obj], km);
+      r = queue_transaction(store, ch, std::move(t));
+      dur += ceph_clock_now() - start;
+
+      ASSERT_EQ(r, 0);
+    }
+    min = (dur < min) ? dur : min;
+    max = (dur > max) ? dur : max;
+    total += dur;
+  }
+  avg = total / obj_count;
+  cout << "[OMAPBENCH] omap_setkeys - min: " << min << " avg: " << avg
+       << " max: " << max << " total: " << total << std::endl;
+
+  // Bench omap_get
+  min = std::numeric_limits<float>::max();
+  max = std::numeric_limits<float>::min();
+  avg = std::numeric_limits<float>::min();
+  total = std::numeric_limits<float>::min();
+
+  cout << "[OMAPBENCH] *** Running omap_get test ***" << std::endl;
+  for (int obj = 0; obj < obj_count; obj++) {
+    bufferlist h;
+    map<string,bufferlist> rm;
+    const utime_t start = ceph_clock_now();
+    store->omap_get(ch, objects[obj], &h, &rm);
+    float dur = ceph_clock_now() - start;
+    min = (dur < min) ? dur : min;
+    max = (dur > max) ? dur : max;
+    total += dur;
+
+    ASSERT_EQ(rm.size(), keys_per_object);
+  }
+  avg = total / obj_count;
+  cout << "[OMAPBENCH] omap_get - min: " << min << " avg: " << avg
+       << " max: " << max << " total: " << total << std::endl;
+
+
+  // Bench seek_to_first
+  min = std::numeric_limits<float>::max();
+  max = std::numeric_limits<float>::min();
+  avg = std::numeric_limits<float>::min();
+  total = std::numeric_limits<float>::min();
+
+  cout << "[OMAPBENCH] *** Running seek_to_first iteration test ***" << std::endl;
+  for (int obj = 0; obj < obj_count; obj++) {
+    map<string,bufferlist> r;
+    ObjectMap::ObjectMapIterator iter = store->get_omap_iterator(ch, objects[obj]);
+    const utime_t start = ceph_clock_now();
+    for (iter->seek_to_first(); iter->valid(); iter->next()) {
+      r[iter->key()] = iter->value();
+    }
+    float dur = ceph_clock_now() - start;
+    min = (dur < min) ? dur : min;
+    max = (dur > max) ? dur : max;
+    total += dur;
+
+    ASSERT_EQ(r.size(), keys_per_object);
+  }
+  avg = total / obj_count;
+  cout << "[OMAPBENCH] seek_to_first - min: " << min << " avg: " << avg
+       << " max: " << max << " total: " << total << std::endl;
+
+  // Bench lower_bound
+  min = std::numeric_limits<float>::max();
+  max = std::numeric_limits<float>::min();
+  avg = std::numeric_limits<float>::min();
+  total = std::numeric_limits<float>::min();
+
+  cout << "[OMAPBENCH] *** Running lower_bound iteration test ***" << std::endl;
+  for (int obj = 0; obj < obj_count; obj++) {
+    map<string,bufferlist> r;
+    ObjectMap::ObjectMapIterator iter = store->get_omap_iterator(ch, objects[obj]);
+    const utime_t start = ceph_clock_now();
+    for (iter->lower_bound(string()); iter->valid(); iter->next()) {
```

Comment:
How much is lower_bound("") similar to first() ?
Maybe lower_bound(dict[dist(gen)]) ?

Reply:
I'm open.  On one hand we could see if it performs differently than seek_to_first() with lower_bound(string()).  There's some evidence that the performance isn't exactly the same.  On the other hand we could make it into a different test based on your idea here.  I'm open to both.